### PR TITLE
Completed array-manipulation-1 exercise.

### PR DESCRIPTION
**Pull request by vkahlon**
cd194e7d1af91bc59ca398fffee08066

### CODE REVIEW

File: array-manipulation-1/src/compact.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+// once the iteration is completed, return the value of the new compactArray.
+ function compact (array) {
+   var compactArray = [];
+   for  (var i = 0; i < array.length; i++) {
+    if (array[i]) {
+      compactArray.push(array[i]);
+    }
+     }
+     return compactArray;
+   }
```
Comment (codingcodymiller) [inline array-manipulation-1/src/compact.js:16]: Whoops, you should fix up your indentation here 😬

File: array-manipulation-1/src/initial.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+// First, a storage space list will need to be allocated in order to not modify the original input.
+// An iteration will be utilized, it will run until it reaches the list last item index.
+// This is accomplished by taking the array length and subtracting it by one.
+// through each iteration, the list item will be copied and placed into the storage space list.
+// After the iteration is completed, it will return the storage space list, that we allocated.
+function initial(array) {
+  var newArray = [];
+  for (var i = 0; i < array.length - 1; i++) {
+    newArray.push(array[i]);
+  } return newArray;
```
Comment (codingcodymiller) [inline array-manipulation-1/src/initial.js:12]: Please always keep the `return` expression on its own line, do not have it on the same line as a closing brace.

File: array-manipulation-1/src/reverse.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+/* exported reverse */
+// In this procedure, we will get a list and return the list but in reverse.
+// we need storage space for the new backwards List
+// We will set up an iteration, and it will run decrementally until the first item of the list
+// The iteration will end when the i reaches an -1 value.
+// each iteraton, it will push the value of the index into the new list.
+// once the iteration is over, it will return the value of the new list.
+function reverse(array) {
+var newArray = [];
+  {
```
Comment (codingcodymiller) [inline array-manipulation-1/src/reverse.js:10]: What is the purpose of this code block?

File: array-manipulation-1/src/tail.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+// To accomplish this, in order to not modify the original input list.
+// We will allocate storage for the new list
+// We will go through an iteration, that takes the lists length, and uses it
+// to find the 2nd item in the list and so forth till the list ends.
+function tail(array) {
+var newArray = [];
+for (var i = 1; i < array.length; i++) {
+  newArray.push(array[i]);
+
+} return newArray;
```
Comment (codingcodymiller) [inline array-manipulation-1/src/tail.js:12]: Please fix your indentation and return statement placement

**Review by codingcodymiller — CHANGES_REQUESTED** [review]

**Comment by vkahlon** [comment]
Those were major formatting mistakes! Thanks for the support! eslint won't always be there to save us.

**Review by codingcodymiller — APPROVED** [review]

**Pull request merged**